tests/v2/validation/upgrade/cloud_provider_aws_migration_test.go

```diff
@@ -0,0 +1,55 @@
+//go:build validation || extended
+
+package upgrade
+
+import (
+	"testing"
+
+	"github.com/rancher/shepherd/clients/rancher"
+	"github.com/rancher/shepherd/extensions/clusters"
+	"github.com/rancher/shepherd/pkg/session"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+const (
+	providerLabel = "provider.cattle.io"
+)
+
+type MigrateCloudProviderSuite struct {
+	suite.Suite
+	session  *session.Session
+	client   *rancher.Client
+	clusters []string
+}
+
+func (u *MigrateCloudProviderSuite) TearDownSuite() {
+	u.session.Cleanup()
+}
+
+func (u *MigrateCloudProviderSuite) SetupSuite() {
+	testSession := session.NewSession()
+	u.session = testSession
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(u.T(), err)
+
+	u.client = client
+}
+
+func (u *MigrateCloudProviderSuite) TestAWS() {
+	_, steveClusterObject, err := clusters.GetProvisioningClusterByName(u.client, u.client.RancherConfig.ClusterName, fleetNamespace)
+	if err != nil {
+		u.Run("RKE1", func() {
+			rke1AWSCloudProviderMigration(u.T(), u.client, u.client.RancherConfig.ClusterName)
+		})
+	} else {
+		u.Run("RKE2", func() {
```

Comment:
Does K3s not support this? Only RKE1 and RKE2? Or is it the case that RKE2 is more just covering v2prov? If so, can we update the name throughout to reflect that?

Reply:
hmm.  I think it would be better in a readme? What do you think; Its clear in the validation/provisioning packages because there's no k3s test for it. Because k3s doesn't support cloud providers out of the box.

Reply:
I'm fine with how it is, I personally was just unaware K3s doesn't support. Thanks for clarifying!